app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :user, foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+
+  def update_posts_counter
+    update(comments_counter: comments.count)
+  end
+
+  def recent_comments(limit = 5)
+    comments.order(created_at: :desc).limit(limit)
+  end
+end
```

Comment:
- [ ] The `belong_to` should belong to `author` and the author is associated to User table cause in the table you have author_id not user_id, it can be similar to 👉🏾 `belongs_to :author, class_name: 'User'`. Kindly fix that 😁 

- [OPTIONAL] I think it would be nice to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to `database`, then make the `update_post_counter` method private 🙃 🙃 
The code can be similar to 👇 

```ruby
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```